src/main/java/com/thegrizzlylabs/sardineandroid/impl/RequestBodyUtil.java

```diff
@@ -27,18 +45,24 @@ public MediaType contentType() {
             @Override
             public long contentLength() {
                 try {
-                    return inputStream.available();
+                    init();
```

Comment:
Does this really return the content length? I would return -1 like in the second snippet of this comment: https://github.com/square/okhttp/issues/3585#issuecomment-327319196
This way you can skip the `init` and open the input stream directly in `Okio.source`.